lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1809,17 +1809,122 @@ bool SwiftLanguageRuntimeImpl::IsValidErrorValue(ValueObject &in_value) {
   return true;
 }
 
+static bool
+CheckIfAnyTypesDeclaredInLLDB(const swift::reflection::TypeRef *typeref) {
+  using swift::Demangle::Node;
+
+  const auto *protocol_typeref = llvm::dyn_cast_or_null<
+      const swift::reflection::ProtocolCompositionTypeRef>(typeref);
+
+  assert(protocol_typeref);
+  if (!protocol_typeref) {
+    assert(false);
+    return false;
+  }
+
+  swift::Demangle::Demangler dem;
+
+  NodePointer type = protocol_typeref->getDemangling(dem);
+  if (type->getKind() != Node::Kind::Type || type->getNumChildren() == 0)
+    return false;
+
+  NodePointer maybe_protocol_list = type->getFirstChild();
+  NodePointer protocol_list;
+
+  if (maybe_protocol_list->getKind() == Node::Kind::ProtocolList)
+    protocol_list = maybe_protocol_list;
+  else if ((maybe_protocol_list->getKind() ==
+                Node::Kind::ProtocolListWithAnyObject ||
+            maybe_protocol_list->getKind() ==
+                Node::Kind::ProtocolListWithClass) &&
+           maybe_protocol_list->getNumChildren() > 0)
+    protocol_list = maybe_protocol_list->getFirstChild();
+  else
+    return false;
+
+  if (protocol_list->getKind() != Node::Kind::ProtocolList ||
+      protocol_list->getNumChildren() == 0)
+    return false;
+
+  NodePointer type_list = protocol_list->getFirstChild();
+  if (type_list->getKind() != Node::Kind::TypeList)
+    return false;
+
+  for (size_t i = 0; i < type_list->getNumChildren(); ++i) {
+    NodePointer internal_type = type_list->getChild(i);
+    if (internal_type->getKind() != Node::Kind::Type ||
```

Comment:
Let's make sure to add a `!internal_type` check here and everywhere else where we call getChild()

Reply:
This shouldn't be necessary as I'm already checking if the parent has the correct amount of children.